integration_tests/src/main/python/json_test.py

```diff
@@ -170,17 +173,33 @@ def test_json_ts_formats_round_trip(spark_tmp_path, date_format, ts_part, v1_ena
     'nan_and_inf.json',
     pytest.param('nan_and_inf_edge_cases.json', marks=pytest.mark.xfail(reason='https://github.com/NVIDIA/spark-rapids/issues/4646')),
     'floats.json',
+    pytest.param('floats_leading_zeros.json', marks=pytest.mark.xfail(reason="cudf always strips off leading zeros")),
```

Comment:
Thanks @revans2 for reviewing this. The reason I had put the xfail is the test fails when those options are not set to `true`. Plugin behavior is same as Spark when the option is set to `true` so those test will get XPASS. 
If the option is `false`(default), it's when it would XFAIL:
- `allowNumericLeadingZeros` - fails on CPU  with an exception where as GPU parses it correctly.
- `allowUnquotedControlChars` - fails on both CPU and GPU.

I was testing with both the options (true, false). Would it be okay to add tests only with `true` and document the default behavior(false) in the .md file below?

Reply:
Yes I am fine with a test that is only run when `allowNumericLeadingZeros` is true.